### PR TITLE
Add the config control to toggle the ability of disabling upload

### PR DESCRIPTION
**Pull request by ShellyWEI**
This allows we control and rollout restrictions inside platform gradually and independently to control the effect to users. azkaban.disable.adhoc.upload is set, we can only disallow the adhoc uploads but it would not affect the lock status being updated at the project level.

### CODE REVIEW

**Comment by tsangz2013** [comment]
This flag impacts the whole webserver that all the users are subjected to it, right?

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java [hunk 1]
```diff
@@ -1965,8 +1967,11 @@ private void ajaxHandleUpload(final HttpServletRequest req, final HttpServletRes
     if (project == null) {
       return;
     }
+
     // fail the upload if the project is UPLOAD locked and the user is not the upload privilege user
-    if (uploadPrivilegeUser != null && project.isUploadLocked() && !user.getUserId().equals(uploadPrivilegeUser)) {
+    // and the cluster-wide disableAdhocUpload flag is set
+    if (uploadPrivilegeUser != null && disableAdhocUploadClusterFlag
```
Comment (svnarumugam) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java:1973]: [Clarification] It appears to me that `lockdownUploadProjects` flag can be used in place of this. Could you help me understand why we need this flag in this place?

Comment (ShellyWEI) [reply azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java:1973]: We don't want to misuse the flags while they appear to have different meanings.
"lockdownUploadProjects" is disallow **ALL** the uploads across the cluster unless it is from admin users and users with special permissions.
"disableAdhocUploadClusterFlag" is disallow the "**upload locked**" projects with the ability to manually upload even if Admin.
I can not apply "lockdownUploadProjects" which would put the restriction on all projects while my intention is to only target "LOCKED" projects.

Comment (svnarumugam) [reply azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java]: Thanks for the explanation @ShellyWEI. Clearer now!

**Pull request merged**